File: lesson_5/code/strange_attractor.py
import matplotlib.pyplot as plt
import numpy as np
from scipy.integrate import solve_ivp


sigma = 10
beta = 8 / 4
rho = 29


def system(_, a: np.ndarray) -> list[float]:
    return [
        -sigma * a[0] + sigma * a[1],
        rho * a[0] - a[1] - a[0] * a[2],
        -beta * a[2] + a[0] * a[1],
    ]


solution = solve_ivp(system, (0, 100), [1, 1, 1], max_step=0.01)
t = solution.t
a = solution.y.T

fig = plt.figure(1)
ax = fig.add_subplot(111, projection="3d")
ax.plot(a[:, 0], a[:, 1], a[:, 2])

plt.figure(2)
plt.plot(t, a[:, 1])
plt.show()

File: lesson_5/code/strange_attractor.m
sigma = 10;
beta = 8/4;
rho = 29;
f = @(t,a) [-sigma*a(1) + sigma*a(2); rho*a(1) - a(2) - a(1)*a(3); -beta*a(3) + a(1)*a(2)];
[t,a] = ode45(f,[0 100],[1 1 1]);     % Runge-Kutta 4th/5th order ODE solver

figure(1)
plot3(a(:,1),a(:,2),a(:,3))

figure(2)
plot(t,a(:,2))


